UM-USElab/gradecraft-development
# Export assignment submissions in a specified date range in the course data exports page

### Status
**IN DEVELOPMENT**

### Description
* When exporting assignment submissions for some courses, the data to export is too large and thus it is useful to specify a date range which can be used as a filter to specify which assignment submissions to export. 
* Now, a form has been added in the /downloads page for the "Assignment Submissions" download link to download only a specified date range if desired (or all the data if no dates are specified)

### Migrations
NO

### Steps to Test or Reproduce
1. As an instructor, visit the /downloads page, and for the "Assignment Submissions" link, specify a date range and a column to apply the date range to
2. Click the link and open the attachment in the email. The data exported should be within the date range for the specified column

### Impacted Areas in Application
* Course data exports (i.e. /downloads)
* Added a script for the form for the "Assignment Submissions" link (assets/javascripts/downloads.js)
* SubmissionExporter now uses a date range
======================
Closes #4315
